File: common/src/main/java/dev/number6/db/dagger/DatabasePortModule.java
package dev.number6.db.dagger;

import dev.number6.db.adaptor.DynamoDatabaseAdaptor;
import dev.number6.db.port.DatabaseConfigurationPort;
import dev.number6.db.port.DatabasePort;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
import dagger.Module;
import dagger.Provides;

@Module(includes = {
        DatabaseConfigurationPortModule.class
})
public class DatabasePortModule {

    @Provides
    public DatabasePort dbService(DynamoDBMapper mapper, DatabaseConfigurationPort dbConfig) {
        return new DynamoDatabaseAdaptor(mapper, dbConfig);
    }

}

File: common/src/main/kotlin/dev/number6/db/dagger/DatabasePortModule.kt
package dev.number6.db.dagger

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
import dagger.Module
import dagger.Provides
import dev.number6.db.adaptor.DynamoDatabaseAdaptor
import dev.number6.db.port.DatabaseConfigurationPort
import dev.number6.db.port.DatabasePort

@Module(includes = [DatabaseConfigurationPortModule::class])
class DatabasePortModule {
    @Provides
    fun dbService(mapper: DynamoDBMapper, dbConfig: DatabaseConfigurationPort): DatabasePort {
        return DynamoDatabaseAdaptor(mapper, dbConfig)
    }
}